### PR TITLE
SINGA-503 Add GlobalAVGPool operator for autograd and onnx

### PR DESCRIPTION
**Pull request by joddiy**
Add GlobalAVGPool operator for autograd and onnx

### CODE REVIEW

File: python/singa/autograd.py [hunk 1]
```diff
@@ -2760,3 +2760,42 @@ def backward(self, dy):
...
+            self.shape_divisor = 1/np.prod(shape[1:-1])
+
+        # output shape
+        # (N x C x 1 x 1) for channels_first
+        # (N x 1 x 1 x C) for channels_last
+        for i in axes:
+            shape[i] = 1
+
+        x = tensor.from_raw_tensor(x)
+        x = tensor.sum(x, axis=axes)
```
Comment (nudles) [inline python/singa/autograd.py]: the raw tensor should also have the sum function?

File: test/python/test_operation.py [hunk 2]
```diff
@@ -2745,5 +2745,97 @@ def test_prelu_broadcast_cpu(self):
...
+        dx = result.creator.backward(dy.data)
+
+        DX = np.ones(X.shape, dtype=np.float32)
+        DX = np.multiply(DX, DY) / np.prod(X.shape[1:-1])
+
+        np.testing.assert_array_almost_equal(tensor.to_numpy(result), XT, decimal=5)
+        np.testing.assert_array_almost_equal(tensor.to_numpy(tensor.from_raw_tensor(dx)), DX, decimal=5)
+
+    def test_globalaveragepool_gpu(self):
+        X = np.array([[[
```
Comment (nudles) [inline test/python/test_operation.py]: can you merge the code for gpu and cpu test?
many lines are repeated..

File: python/singa/autograd.py [hunk 3]
```diff
@@ -2760,3 +2760,42 @@ def backward(self, dy):
...
+
+
+class GlobalAveragePool(Operation):
+    def __init__(self, data_format='channels_first'):
+        super(GlobalAveragePool, self).__init__()
+        self.data_format = data_format
+
+    def forward(self, x):
+        if training:
+            self.mask = singa.Tensor(list(x.shape()), x.device())
```
Comment (nudles) [inline python/singa/autograd.py]: can remove list()?

**Comment by nudles** [comment]
need to add some docstrings for the new operators.

File: test/python/test_operation.py (hunk 2 above)
Comment (nudles) [reply test/python/test_operation.py]: like this [test](https://github.com/apache/singa/pull/515/files#diff-c0a014dc6b3a07acd5e7c10001de82c9R1483)

Comment (joddiy) [reply test/python/test_operation.py]: fixed

File: python/singa/autograd.py (hunk 3 above)
Comment (joddiy) [reply python/singa/autograd.py]: fixed

File: python/singa/autograd.py (hunk 1 above)
Comment (joddiy) [reply python/singa/autograd.py]: it seems the singa.Sum does not support sum over axis larger than 1.

**Comment by joddiy** [comment]
> need to add some docstrings for the new operators.

fixed

**Pull request merged**